### ISSUE TITLE
Leverage Circuit Providers instead of your own Service Provider Model

### ISSUE THREAD

**Issue by ryanmerolle** [mlebreuil/netbox-contract#42]
It gets super confusing when you need to duplicate your Circuit Providers into the plugin's Service Provider model.

**Comment by ryanmerolle** [comment]
I just spotted this on the README:

```markdown
The pluggin adds contracts and invoices model to Netbox.
It allows to register contract with objects.
Add invoices to contracts.
```

I guess you are looking to track "contracts" against other objects besides circuits in the long term?  I raised this issue because it seemed like a focus on circuits.  I was going to suggest a plugin rename to focus on circuit billing or circuit contracts.

**Comment by mlebreuil** [comment]
Sorry for the late reply. Indeed the aim is to be able to attache contracts to multiple object types, not only Circuits.

**Comment by mlebreuil** [comment]
That said, I understand the confusion. Not sure how to best handle this.

**Comment by ryanmerolle** [comment]
so you could setup a GenericForeignKey or some other option to leverage to link existing models or existing providers/tenants/etc to a contract.  You could have the GFK be limited to your plugin model's provider and the circuit provider models.

**Comment by mlebreuil** [comment]
Closed with #100
